src/main/java/com/example/andersen/Task6/bucket/BucketImpl.java

```diff
@@ -0,0 +1,153 @@
+package com.example.andersen.Task6.bucket;
+
+import com.example.andersen.Task5.exception.PutWrongNumberException;
+import com.example.andersen.Task6.dao.Product;
+import com.example.andersen.Task6.warehouse.Warehouse;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static com.example.andersen.Task6.warehouse.Warehouse.checkForExpiration;
+
+@Service
+public class BucketImpl implements Bucket {
+    private final Warehouse warehouse = new Warehouse();
+    private Map<Product, Integer> orders = new HashMap<>();
+
+    public boolean isEnoughAmountOfProduct(int product_id, int wanted_amount) {
+        Map<Product, Integer> map = warehouse.getAllProducts();
+        Product wanted = findProductById(product_id);
+        if (wanted == null) {
+            throw new PutWrongNumberException(String.valueOf(product_id));
+        }
+        int existAmount = map.get(wanted);
+        return existAmount >= wanted_amount;
+    }
+
+    @Override
+    public void addProduct(int product_id, int addAmount) {
+        if (addAmount <= 0 || !isEnoughAmountOfProduct(product_id, addAmount)) {
```

Comment:
Try to avoid negative expressions. It`s better to create additional method isNotEnoughAmountOfProduct

Reply:
Done